refactor: modify trending browser navigators

Prithpal-Sooriya opened the pull request:
## **Description**

Fixes issue where custom trending browser uses a different design compared to the original.
This PR removes our custom browser wrapper in favour of the original browser + behaviour -- the mobile platform team will spearhead the IAB changes.

For visibility: 

Navigation Structure Before:
```
(TAB) REMOVED Browser Tab -- removing this caused issues for areas using old browser navigation
(TAB) Trending Tab (ExploreFlow)
  (STACK) Trending
    (STACK) Trending
      - TRENDING_FEED
      - EXPLORE_SEARCH
      - Custom Browser
```

Navigation Structure After:

```
(TAB) Browser Tab (BrowserFlow) -- Keeping original browser tab, but it is hidden - existing navigations will still work
  - BROWSER.VIEW
  - BROWSER.ASSET_LOADER
  - BROWSER.ASSET_VIEW
(TAB) Trending Tab (ExploreFlow)
  - TRENDING_FEED
  - EXPLORE_SEARCH
  - Dedicated BrowserFlow screen -- Allows trending page to use existing browser (no custom browser), and retains trending navigation back behaviour (search queries and navigation stack is retained)
```



## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: refactor: modify trending/explore page navigation stacks

## **Related issues**

Fixes:

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

https://www.loom.com/share/bb1343b7a7ed4b1abb841f8c6d1284f3

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.



<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Refactors Explore/Trending to use the main Browser flow (with a hidden Browser tab), adds `TRENDING_FEED`, and enhances TabBar to support custom selection/hidden states, updating routes and tests accordingly.
> 
> - **Navigation / Explore-Trending**:
>   - Replace custom Trending browser with existing `BrowserFlow`; remove `BrowserWrapper` and related screens.
>   - Introduce `ExploreHome` stack with `Routes.TRENDING_FEED`, `Routes.EXPLORE_SEARCH`, `Routes.SITES_FULL_VIEW`, and nested `Routes.BROWSER.HOME`.
>   - Keep `Routes.BROWSER.HOME` as a hidden tab when trending is enabled; Trending tab considers both `TRENDING_VIEW` and `BROWSER.HOME` as selected.
>   - Update `BrowserTab` back behavior to return to `TRENDING_VIEW -> TRENDING_FEED` when not launched from Trending.
>   - Add `Routes.TRENDING_FEED`.
> - **TabBar**:
>   - Add `options.isSelected(rootScreenName)` and `options.isHidden` to `ExtendedBottomTabDescriptor`.
>   - Change Trending tab press to `navigation.reset({ routes: [{ name: Routes.TRENDING_VIEW }] })`.
> - **Sites/Explore interactions**:
>   - `SiteRowItemWrapper` and `SitesSearchFooter` now navigate to `Routes.BROWSER.HOME` with `screen: Routes.BROWSER.VIEW` and `{ fromTrending: true, newTabUrl, timestamp }`.
> - **Tests / Snapshots**:
>   - Update unit tests and snapshots for new navigation targets and removal of deprecated screens (`TrendingBrowser`, `BrowserWrapper`).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ec056126204de16f99411bd68e0fa616ddf63aac. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

Prithpal-Sooriya committed:
refactor - remove custom browser and flatten nested trending routes

Prithpal-Sooriya committed:
test: update UTs

Prithpal-Sooriya commented on app/components/Nav/Main/MainNavigator.js:
```diff
@@ -278,25 +278,6 @@ const RewardsHome = () => (
-      options={{ headerShown: false }}
-    />
-  </Stack.Navigator>
-);
```
With our flat navigation, we don't need any ref updates.

Prithpal-Sooriya committed:
refactor - undo route name changes

we can clean this up as a separate PR

Prithpal-Sooriya committed:
further flatten routing for trending + browser

added hometab overrides to modify selected tab and hide other tabs

Prithpal-Sooriya committed:
revert: remove force FF flag enabled

juanmigdr commented:
I also see now the search screen opens from the bottom instead of from the side, I know design team want to migrate to having the animation from right to left for all screens and this was the previous behavior
https://github.com/user-attachments/assets/8c2c4841-5af9-44a0-8e36-753b690b723d

Prithpal-Sooriya committed:
update routing so that nested trending pages correcly invoke browser

Prithpal-Sooriya committed:
test: update snapshots

Prithpal-Sooriya committed:
Merge branch 'main' of github.com:MetaMask/metamask-mobile into refactor/modify-trending-navigators

Prithpal-Sooriya committed:
feat: update browser stack functionality for trending pages

Prithpal-Sooriya commented on app/components/Nav/Main/MainNavigator.js:
```diff
@@ -324,6 +305,48 @@ const BrowserFlow = (props) => (
   </Stack.Navigator>
 );
 
+const ExploreHome = () => (
```
Explore stack is flat, not additional nesting.

We are also reusing the existing browser instead of a custom browser.
- We need a dedicated browser stack to retain the explore stack (e.g. retain search parameters on the explore search)

Prithpal-Sooriya commented on app/components/Nav/Main/MainNavigator.js:
```diff
@@ -642,10 +665,12 @@ const HomeTabs = () => {
+    const isInBrowser =
+      currentRoute.name?.startsWith(Routes.BROWSER.HOME) ||
+      currentStackRouteName?.startsWith(Routes.BROWSER.HOME);
+    if (isBrowserFullscreen && isInBrowser) {
```
This is the fullscreen bug fix.

We now check route names against the main navigation (e.g. tab navigation) and nested navigation (e.g. stack navigation).

This is because the browser can be opened up in the existing tab, or through trending.

Prithpal-Sooriya committed:
test: fix UTS

Prithpal-Sooriya committed:
style: add trending search anims

juanmigdr reviewed (dismissed):
LGTM, nice one

brianacnguyen reviewed (changes requested):
Question regarding TabBar changes

Prithpal-Sooriya committed:
refactor: update TabBar props

flatten the override props, and rename the new properties to "isSelected" and "isHidden"

Cal-L reviewed (approved):
Lgtm!
